server/core/src/repl/mod.rs

```diff
@@ -363,11 +368,60 @@ async fn repl_run_consumer(
     Some(socket_addr)
 }
 
+#[instrument(level="info", skip(idms), fields(eventid=Uuid::new_v4().to_string()))]
+async fn replication_trigger(
+    idms: &IdmServer,
+    last_replication_instant: &Instant,
+    enable_experimental_replication_trigger: bool,
+    replication_interval: Duration,
+) {
+    tokio::select! {
+        _ = async {
```

Comment:
This should probably be a function instead, it's really hard to parse this.

Reply:
Wopsies hadn't thought of that :p. I was too enthused by finally understanding the whole tokio::select! business :D